src/templates/list-table-block/dialog/unbind-dialog.vue

```diff
@@ -0,0 +1,184 @@
+        </div>
+      </div>
+    </div>
+
+    <template #closeicon>
+      <PrimeButton
+        outlined
+        @click="cancelDialog()"
+        icon="pi pi-times"
+        data-testid="unbind-dialog__close-icon__button"
+      />
+    </template>
+
+    <template #footer>
+      <PrimeButton
+        outlined
+        label="Cancel"
+        @click="cancelDialog()"
+        data-testid="unbind-dialog__footer__cancel-button"
+      />
+      <PrimeButton
+        severity="danger"
+        label="Unbind"
+        icon-pos="right"
+        @click="removeItem()"
+        :icon="getLoadingIcon"
+        :disabled="isDisabled"
+        data-testid="unbind-dialog__footer__unbind-button"
+      />
+    </template>
+  </PrimeDialog>
+</template>
+
+<script setup>
+  import { computed, ref, watch, inject } from 'vue'
+  import { useField, useForm } from 'vee-validate'
+  import { useToast } from 'primevue/usetoast'
+  import * as yup from 'yup'
+  import PrimeButton from 'primevue/button'
+  import PrimeDialog from 'primevue/dialog'
+  import InputText from 'primevue/inputtext'
+  import Message from 'primevue/message'
+
+  const emit = defineEmits(['successfullyUnbound'])
+
+  const toast = useToast()
+  const dialogRef = inject('dialogRef')
+
+  const data = dialogRef.value.data
+  const loading = ref(false)
+  const canUnbind = ref(false)
+
+  const validationSchema = yup.object({
+    confirmation: yup.string().equals(['unbind'], '').required('This is a required field')
+  })
+
+  const { errors, meta, resetForm } = useForm({
+    validationSchema,
+    initialValues: {
+      confirmation: ''
+    }
+  })
+
+  const { value: confirmation } = useField('confirmation')
+
+  const removeItem = async () => {
+    if (!canUnbind.value || !meta.value.valid) return
+
+    loading.value = true
+    try {
+      const feedback = await data.unbindingService(data.selectedID, data.selectedItemData)
+      showToast('success', feedback ?? 'Unbound successfully!')
+      emit('successfullyUnbound')
+      resetForm()
+      dialogRef.value.close({ updated: true })
+    } catch (error) {
+      showToast('error', 'Error', error)
+    } finally {
+      loading.value = false
+    }
+  }
+
+  const showToast = (severity, summary, detail = '') => {
+    toast.add({
+      closable: true,
+      severity,
+      summary,
+      detail
+    })
+  }
+
+  const cancelDialog = () => {
+    resetForm()
+    dialogRef.value.close({ updated: false })
+  }
+
+  const getLoadingIcon = computed(() => {
+    return loading.value ? 'pi pi-spin pi-spinner' : ''
+  })
+
+  const isDisabled = computed(() => {
+    return !meta.value.valid || loading.value
+  })
+
+  watch(
+    () => data,
+    (value) => {
+      if (value) {
+        canUnbind.value = false
+        resetForm()
+      }
+    },
+    { deep: true }
+  )
+
+  watch(
+    () => confirmation.value,
+    (value) => {
+      if (value) {
+        canUnbind.value = true
+      }
+    },
+    { deep: true }
+  )
```

Comment:
Acredito que não haja necessidade de utilizar watch, recomendo deixar btn habilitado é exibir um error no campo como obrigatório.
Se o padrão de delete for disabled, então adicionar o confirmation.value no isDisabled.

Reply:
Resolvido